pkg/drain/nodegroup.go

```diff
@@ -142,12 +140,19 @@ func NodeGroup(clientSet kubernetes.Interface, ng *api.NodeGroup, waitTimeout ti
 			logger.Debug("already drained: %v", drainedNodes.List())
 			logger.Debug("will drain: %v", newPendingNodes.List())
 
+		evict_loop:
 			for _, node := range nodes.Items {
 				if newPendingNodes.Has(node.Name) {
 					pending, err := evictPods(drainer, &node)
 					if err != nil {
 						logger.Warning("pod eviction error (%q) on node %s – will retry after delay of %s", err, node.Name, retryDelay)
-						time.Sleep(retryDelay)
+						retryTimer := time.NewTimer(retryDelay)
```

Comment:
This also sleeps?

Reply:
It doesn't. The receive on the `retryTimer.C` channel below blocks until `retryDelay` elapses.